discord/commands/core.py

```diff
@@ -806,6 +806,19 @@ def _parse_options(self, params, *, check_params: bool = True) -> list[Option]:
             if option == inspect.Parameter.empty:
                 option = str
 
+            if self._is_typing_literal(option):
+                literal_values = get_args(option)
+                if not all(isinstance(v, (str, int, float)) for v in literal_values):
+                    raise TypeError(
+                        "Literal values must be str, int, or float for Discord choices."
+                    )
+                option = Option(
+                    str,
+                    choices=[
+                        OptionChoice(name=str(v), value=str(v)) for v in literal_values
+                    ],
+                )
+
```

Comment:
We shouldn't force all values to be strings, option choice values of int and float are also allowed but only when the option has the matching input type. For that, we'd have to make sure all literal values are of the same type and that the type is either str, int or float. I'm just not sure whether this is the right place to implement this logic since option choices can be defined in other ways too. Either way, I want most of the logic related to options to be refactored for v3.0, therefore I'd rather put this pull request on hold.